### ISSUE TITLE
KeyDB 6.0.13 blocks with Redisearch garbage collector

### ISSUE THREAD

**Issue by francisco-batista-wize** [JohnSully/KeyDB#214]
**Describe the bug**
I am using KeyDB v6.0.13 with Redisearch 1.8.1, it works fine with Redisearch's garbage collector disabled. But the server locks after a coulpe seconds when adding an index and some data. It also blocks on startup while loading an RDB.

**To Reproduce**
To recreate the index creation scenario, run the following commands:
```
> FT.CREATE a SCHEMA b NUMERIC
> FT.ADD a idx 1 FIELDS b 1
```
If you run them with GC enabled, it will lock after 30 seconds, but if GC is disabled, it will work fine.

To recreate the RDB load scenario, disable GC and run the following commands:
```
> FT.CREATE a SCHEMA b NUMERIC
> FT.ADD a idx 1 FIELDS b 1
> SAVE
```
Then, shutdown the instance, enable GC and run the server, it will lock instantly.

---
**For reference**
I also tried KeyDB 6.0.13 with Redisearch 1.6.13 and 1.6.12, it crashes after 30 seconds with GC enabled.
Issue #209 seems to address this issue, but the bug persists in this version

**Comment by adalbertolv** [comment]
I'm facing the same issue, have you found a solution?

**Comment by alain-menchaca-clip** [comment]
I have the same problem. Is there an update here?

**Comment by botzill** [comment]
Same here.

**Comment by btzsoft** [comment]
the same.

**Comment by botzill** [comment]
@francisco-batista-wize How do you disable GC?

**Comment by francisco-batista-wize** [comment]
@botzill Loading the module with the `NOGC` flag

**Comment by botzill** [comment]
Thx @francisco-batista-wize I did that, but still, smth is wrong with `1.99.3`. When I delete a key it just hangs. I tried with `edge` version and seems to be OK, will test more.

**Comment by smellbee** [comment]
this bug still exists in 6.0.16,  when using keydb with redisearch module, we can't add a new hash to an index, it will halt the keydb server entirely

**Comment by VivekSainiEQ** [comment]
Hi all, 

This issue was resolved in 7b4e25008a352bd425582a3e60b26894826af62,

KeyDB was creating module timer events in a way that caused deadlocks previously, which the Redisearch garbage collector creates on load and every 30 seconds afterwards. The timer events have been changed so this deadlock no longer occurs.
